guides/source/active_record_validations.md

```diff
@@ -278,39 +310,71 @@ end
 ```
 
 ```irb
-irb> Person.new.errors[:name].any?
+irb> new_person = Person.new
+irb> new_person.errors[:name]
+=> [] # no errors since validations are not run until saved
+irb> new_person.errors[:name].any?
 => false
-irb> Person.create.errors[:name].any?
+
+irb> saved_person = Person.create
+irb> saved_person.errors[:name]
+=> ["can't be blank"] # validation error because `name` is required
+irb> saved_person.errors[:name].any?
 => true
 ```
 
-We'll cover validation errors in greater depth in the [Working with Validation
-Errors](#working-with-validation-errors) section.
+Additionally, you can use the
+[`errors.add`](https://api.rubyonrails.org/classes/ActiveModel/Errors.html#method-i-add)
+method to manually add error messages for specific attributes. This is
+particularly useful when defining custom validation scenarios.
 
-[Errors#squarebrackets]: https://api.rubyonrails.org/classes/ActiveModel/Errors.html#method-i-5B-5D
+```ruby
+class Person < ApplicationRecord
+  validate do |person|
+    errors.add :name, :too_short, message: "is not long enough"
+  end
+end
+```
+
+NOTE: To read about validation errors in greater depth refer to the [Working
+with Validation Errors](#working-with-validation-errors) section.
+
+[Errors#squarebrackets]:
+    https://api.rubyonrails.org/classes/ActiveModel/Errors.html#method-i-5B-5D
 
 Validation Helpers
 ------------------
```

Comment:
This is an excellent point. I did the following: 

1. I renamed the header to Validations 
2. I added this section to clarify the different notations:

> In modern Rails applications, the more concise validate syntax is commonly used, for example:
> 
> ```
> validate :name, presence: true
> ```
> 
> However, older versions of Rails used "helper" methods, such as:
> 
> ```
> validates_presence_of :name
> ```
> 
> Both notations perform the same function, but the newer form is recommended for its readability and alignment with Rails' conventions.

3. In the info section that links to the API Helpers

> INFO: To see a list of the available default helpers, take a look at
> [`ActiveModel::Validations::HelperMethods`][]. This API section uses the older notation as described above.

4. User "validations" instead of "validation helpers".

See https://github.com/rails/rails/pull/52951/commits/164e19c1d461da9759b9356c8758205533733974